### ISSUE TITLE
iOS cmake build should use a toolchain file

### ISSUE THREAD

**Issue by tlein** [SFML/SFML#1268]
**Problem:** Currently SFML on iOS builds itself with CMake but it does not use a toolchain file. Instead, it sets the necessary CMake variables itself instead of relying on the external toolchain file. It's worth noting that SFML **does** use a toolchain file for android. 

**Suggested Solution:** Ship an iOS toolchain file with SFML and have the user specify CMAKE_TOOLCHAIN_FILE=/path/to/toolchain.cmake instead of IOS=True

**Comment by headupinclouds** [comment]
There is a nice set of toolchains for iOS maintained in [polly](https://github.com/ruslo/polly) (companion project for the [hunter](https://github.com/ruslo/hunter) package manager).  

Random examples:
```
ios-10-3-dep-8-0-libcxx-hid-sections-lto.cmake
ios-10-3-dep-8-0-bitcode.cmake
ios-10-3-armv7.cmake
ios-nocodesign-10-0-arm64.cmake
ios-10-2.cmake
```

**Comment by mantognini** [comment]
Could you motivate the need for such toolchain (that needs to be updated and maintained)? The fact that it's used for android seems irrelevant to me. What is currently broken?

**Comment by tlein** [comment]
Sure! Currently the cmake variables setup in SFML would also have to be updated and maintained. Centralizing iOS logic in the tool chain allows us to simplify the rest of the cmake files. This is the standard pattern when setting up a cmake project for cross compiling.

As for my personal motivation, my team has a build system that pulls down and builds all of our projects dependencies when initiating a build for a given platform and architecture. SFML is one of those dependencies and our build system will blindly set the toolchain file for a cross compiled system. This means all of our dependencies are built with the same toolchain file. It would be a hassle to work around specific dependencies quirks when there is a standard pattern cmake already has.

**Comment by mantognini** [comment]
> Centralizing iOS logic in the tool chain allows us to simplify the rest of the cmake files. 

Seems like a reasonable thing to do, indeed.

**Comment by eXpl0it3r** [comment]
Fixed with #1269
